webserver.h

```diff
@@ -43,7 +43,38 @@ class WebServer
 
     void Start()
     {
-        // Define the `/api/canvases` endpoint
+        // Define the `/api/sockets` endpoint
+        CROW_ROUTE(_crowApp, "/api/sockets")
+            .methods(crow::HTTPMethod::GET)([&]() -> crow::response
+            {
+                auto socketsJson = nlohmann::json::array();
+                for (size_t canvasId = 0; canvasId < _allCanvases.size(); ++canvasId)
+                {
+                    if (_allCanvases[canvasId]) // Ensure canvas exists
+                    {
+                        for (size_t featureId = 0; featureId < _allCanvases[canvasId]->Features().size(); ++featureId)
+                        {
+                            const auto& feature = _allCanvases[canvasId]->Features()[featureId];
+                            nlohmann::json socketJson;
+                            socketJson["hostName"] = feature->Socket().HostName();
+                            socketJson["friendlyName"] = feature->Socket().FriendlyName();
+                            socketJson["featureId"] = featureId;
+                            socketJson["canvasId"] = _allCanvases[canvasId]->Id();
+                            socketJson["isConnected"] = feature->Socket().IsConnected();
+                            socketJson["bytesPerSecond"] = feature->Socket().BytesSentPerSecond();
+                            socketJson["port"] = feature->Socket().Port();
+                            socketsJson.push_back(socketJson);
+                        }
+                    }
+                }
+                return socketsJson.dump();
+            });
+
+            // Define the `/api/sockets/:id` endpoint
+            CROW_ROUTE(_crowApp, "/api/sockets/<int>")
+                .methods(crow::HTTPMethod::GET)([&](int id) -> crow::response {
+
+                });        // Define the `/api/canvases` endpoint
```

Comment:
Maybe this is still work in progress, but this endpoint's lambda now doesn't contain any code, meaning it won't return anything. I'm not even sure that compiles, considering the lambda promises to produce a `crow::response` instance?

Reply:
It's "TODO" yet

Reply:
That's obviously fair enough. I'll revisit this when it's no longer on that list. :)